src/Service/DynamoDb/src/Result/QueryOutput.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+namespace AsyncAws\DynamoDb\Result;
+
+use AsyncAws\Core\Result;
+use AsyncAws\DynamoDb\DynamoDbClient;
+use AsyncAws\DynamoDb\Input\QueryInput;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+class QueryOutput extends Result implements \IteratorAggregate
+{
+    private $Items = [];
+
+    private $Count;
+
+    private $ScannedCount;
+
+    private $LastEvaluatedKey = [];
+
+    private $ConsumedCapacity;
+
+    /**
+     * The capacity units consumed by the `Query` operation. The data returned includes the total provisioned throughput
+     * consumed, along with statistics for the table and any indexes involved in the operation. `ConsumedCapacity` is only
+     * returned if the `ReturnConsumedCapacity` parameter was specified. For more information, see Provisioned Throughput in
+     * the *Amazon DynamoDB Developer Guide*.
+     *
+     * @see https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/ProvisionedThroughputIntro.html
+     */
+    public function getConsumedCapacity(): ?ConsumedCapacity
+    {
+        $this->initialize();
+
+        return $this->ConsumedCapacity;
+    }
+
+    /**
+     * The number of items in the response.
+     */
+    public function getCount(): ?int
+    {
+        $this->initialize();
+
+        return $this->Count;
+    }
+
+    /**
+     * @param bool $currentPageOnly When true, iterates over items of the current page. Otherwise also fetch items in the next pages.
+     *
+     * @return iterable<AttributeValue[]>
+     */
+    public function getItems(bool $currentPageOnly = false): iterable
+    {
+        if ($currentPageOnly) {
+            $this->initialize();
+            yield from $this->Items;
+
+            return;
+        }
+
+        $client = $this->awsClient;
+        if (!$client instanceof DynamoDbClient) {
+            throw new \InvalidArgumentException('missing client injected in paginated result');
+        }
+        if (!$this->input instanceof QueryInput) {
+            throw new \InvalidArgumentException('missing last request injected in paginated result');
+        }
+        $input = clone $this->input;
+        $page = $this;
+        while (true) {
+            if ($page->getLastEvaluatedKey()) {
+                $input->setExclusiveStartKey($page->getLastEvaluatedKey());
```

Comment:
This is similar to what @jderusse and I talked about the other day, 

> Argument 1 of AsyncAws\DynamoDb\Input\QueryInput::setExclusiveStartKey expects array<array-key, AsyncAws\DynamoDb\Input\AttributeValue>, array<array-key, AsyncAws\DynamoDb\Result\AttributeValue> provided

Reply:
should be fixed by #316